chrome/content/zotero/reactUI/containers/tagSelector.jsx

```diff
@@ -0,0 +1,411 @@
+/* global Zotero: false */
+'use strict';
+
+(function() {
+
+const React = require('react');
+const ReactDom = require('react-dom');
+const { IntlProvider } = require('react-intl');
+const TagSelector = require('react-ui/components/tag-selector.js');
+const noop = Promise.resolve();
+const defaults = {
+	tagColors: new Map(),
+	tags: [],
+	showAutomatic: Zotero.Prefs.get('tagSelector.showAutomatic'),
+	searchString: '',
+	inScope: new Set(),
+};
+const { Cc, Ci } = require('chrome');
+
+ZoteroPane.React.TagSelector = class TagSelectorContainer extends React.Component {
+	constructor(props) {
+		super(props);
+		this._notifierID = Zotero.Notifier.registerObserver(
+			this,
+			['collection-item', 'item', 'item-tag', 'tag', 'setting'],
+			'tagSelector'
+		);
+		this.filterToScope = Zotero.Prefs.get('tagSelector.filterToScope');
+		this.selectedTags = new Set();
+		this.state = defaults;
+		this.onResize();
+	}
+
+	// Update trigger #1 (triggered by ZoteroPane)
+	async onItemViewChanged({collectionTreeRow, libraryID, tagsInView}) {
```

Comment:
So this is clearly not The React Way. Is the goal just to minimize code changes until we have Redux or some other sort of global state mechanism? Any reason ZoteroPane can't just set these as props on the tag selector component? (Or @tnajdek might have thoughts on using an interim global state object before we get to full-blown Redux.)

`notify()` below is a bit trickier, since that's tied to `Zotero.Notifier`. So leaving that as-is for now might make sense, until we figure out a better way to do that everywhere.